test/example/type-builder-nominal-new.cpp

```diff
@@ -0,0 +1,435 @@
+#include <cassert>
+#include <iostream>
+
+#include "wasm-type.h"
+
+/* This file is identical to type-builder-nominal.cpp, except that instead of
+ * setting the global type system to be nominal, it instead individually sets
```

Comment:
Diffing the output, there are a few changes, like:
```diff
 Before setting heap types:
-(ref $sig) => (ref [T](func))
-(ref $struct) => (ref [T](func))
-(ref $array) => (ref [T](func))
-(ref null $array) => (ref null [T](func))
-(rtt 0 $array) => (rtt 0 [T](func))
+(ref $sig) => [T](ref [T](func))
+(ref $struct) => [T](ref [T](func))
+(ref $array) => [T](ref [T](func))
+(ref null $array) => [T](ref null [T](func))
+(rtt 0 $array) => [T](rtt 0 [T](func))
```
Is that intentional to indicate a milestone4 type?

Reply:
The `[T]` signifies a temporary type or heap type. In `--nominal` mode, the "temporary" types constructed by a `TypeBuilder` are actually allocated in the global store, so they are not truly temporary. But in equirecursive canonicalization, temporary types are in fact initially temporary, and that difference is reflected here.